From 501bd08ac3aae2a5115ca37e719741c33ed224a6 Mon Sep 17 00:00:00 2001
From: ManojBogandham <manojbvs01@gmail.com>
Date: Thu, 25 Jun 2020 19:42:06 +0530
Subject: [PATCH] 'client3'

---
 Task1/client3.py | 9 ++++++---
 1 file changed, 6 insertions(+), 3 deletions(-)

diff --git a/Task1/client3.py b/Task1/client3.py
index f1771c3..354f0cc 100644
--- a/Task1/client3.py
+++ b/Task1/client3.py
@@ -35,14 +35,16 @@ def getDataPoint(quote):
 	stock = quote['stock']
 	bid_price = float(quote['top_bid']['price'])
 	ask_price = float(quote['top_ask']['price'])
-	price = bid_price
+	price = (bid_price+ask_price)/2
 	return stock, bid_price, ask_price, price
 
 def getRatio(price_a, price_b):
 	""" Get ratio of price_a and price_b """
 	""" ------------- Update this function ------------- """
 	""" Also create some unit tests for this function in client_test.py """
-	return 1
+	if(price_b==0)
+		return
+	return (price_a/price_b)
 
 # Main
 if __name__ == "__main__":
@@ -54,6 +56,7 @@ if __name__ == "__main__":
 		""" ----------- Update to get the ratio --------------- """
 		for quote in quotes:
 			stock, bid_price, ask_price, price = getDataPoint(quote)
+			prices[stock]=price
 			print ("Quoted %s at (bid:%s, ask:%s, price:%s)" % (stock, bid_price, ask_price, price))
 
-		print ("Ratio %s" % getRatio(price, price))
+		print ("Ratio %s" % getRatio(prices['ABC'], prices['DEF']))
-- 
2.27.0.windows.1

